packages/core/src/hooks/auth/usePermissions/index.ts

```diff
@@ -87,7 +90,7 @@ export function usePermissions<TData = any>({
             "v3LegacyAuthProviderCompatible",
         ],
         // Enabled check for `getPermissions` is enough to be sure that it's defined in the query function but TS is not smart enough to know that.
-        queryFn: legacyGetPermission ?? (() => Promise.resolve(undefined)),
+        queryFn: (legacyGetPermission ? () => legacyGetPermission(params) : () => Promise.resolve(undefined)) as (params?: unknown) => Promise<TData>,
```

Comment:
```suggestion
        queryFn: (legacyGetPermission ? () => legacyGetPermission(params) : () => Promise.resolve(undefined)) as (params?: TParams) => Promise<TData>,
```

I think we can use `TParams` here since.

Reply:
@BatuhanW  I tried replacing unknown to TParams, I got this error `Object literal may only specify known properties, and 'meta' does not exist in type 'readonly unknown[]'`, so I had to leave it as unknown